gcloud-java-examples/src/main/java/com/google/gcloud/examples/DnsExample.java

```diff
@@ -0,0 +1,516 @@
+      } else {
+        System.out.println("Project contains no zones.");
+      }
+    }
+
+    @Override
+    public String params() {
+      return "";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 0;
+    }
+  }
+
+  private static class GetZoneAction implements DnsAction {
+
+    /**
+     * Gets details about a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      Zone zone = dns.getZone(zoneName);
+      if (zone == null) {
+        System.out.printf("No zone with name '%s' exists.%n", zoneName);
+      } else {
+        System.out.printf("Name: %s%n", zone.name());
+        System.out.printf("ID: %s%n", zone.id());
+        System.out.printf("Description: %s%n", zone.description());
+        System.out.printf("Created: %s%n", zone.creationTimeMillis());
+        System.out.printf("Name servers: %s%n", Joiner.on(",").join(zone.nameServers()));
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+  }
+
+  private static class DeleteZoneAction implements DnsAction {
+
+    /**
+     * Deletes a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      boolean deleted = dns.delete(zoneName);
+      if (deleted) {
+        System.out.printf("Zone %s was deleted.%n", zoneName);
+      } else {
+        System.out.printf("Zone %s was NOT deleted. It probably does not exist.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+
+  }
+
+  private static class DeleteDnsRecordAction implements DnsAction {
+
+    /**
+     * Deletes a DNS record of type A from the given zone. The last parameter is ttl and it is not
+     * required.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String recordName = args[1];
+      String ip = args[2];
+      DnsRecord record = DnsRecord.builder(recordName, DnsRecord.Type.A)
+          .records(ImmutableList.of(ip))
+          .build();
+      if (args.length > 3) {
+        Integer ttl = Integer.valueOf(args[3]);
+        record = record.toBuilder().ttl(ttl, TimeUnit.SECONDS).build();
+      }
+      ChangeRequest changeRequest = ChangeRequest.builder()
+          .delete(record)
+          .build();
+      changeRequest = dns.applyChangeRequest(zoneName, changeRequest);
+      System.out.printf("The request for deleting A record %s for zone %s was successfully " +
+          "submitted and assigned ID %s.%n", recordName, zoneName, changeRequest.id());
+      while (changeRequest.status().equals(ChangeRequest.Status.PENDING)) {
```

Comment:
If there's a significant hang, it'd be nice to print something periodically to terminal so that users know the dns service is working hard :)